controllers/managedocs_controller_test.go

```diff
@@ -1114,6 +1187,115 @@ var _ = Describe("ManagedOCS controller", func() {
 				}, timeout, interval).Should(Equal(ctrlutils.GetResourceRequirements("ocs-operator")))
 			})
 		})
+		When("there is a notification email address in the add-on parameter secret and smtp secret", func() {
+			It("should update alertmanager email config with smtp details", func() {
+				setupAlertmanagerConfigConditions(true, "test-key", true, "test-url", true, []string{"test@email.com"})
+
+				// Wait for alertmanager to get updated with smtp details
+				amconfig := amConfigTemplate.DeepCopy()
+				amconfigKey := utils.GetResourceKey(amconfig)
+				var resIndex int
+				Eventually(func() bool {
+					Expect(k8sClient.Get(ctx, amconfigKey, amconfig)).Should(Succeed())
+					for i, reciever := range amconfig.Spec.Receivers {
+						if reciever.Name == "SendGrid" {
+							resIndex = i
+						}
+					}
+					return len(amconfig.Spec.Receivers[resIndex].EmailConfigs) > 0
+				}, timeout, interval).Should(Equal(true))
```

Comment:
A couple of comments:
1. This for loop will create an in-memory copy of the receiver.
2. You do not need to wait for the loop to end after finding the proper receiver. 
3. Why are we checking the length of the `EmialConfigs`? this test case is not about the email config it is about the existence of SMTP receiver. The tests below already cover the email configs.

The suggestion below will fix both these issues:
```suggestion
				Eventually(func() bool {
					for i := range amconfig.Spec.Receivers {
						reciever := &amconfig.Spec.Receivers[i]
						if reciever.Name == "SendGrid" {
							return true;
						}
					}
				}, timeout, interval).Should(Equal(true))
```

On top of that this `Eventually` construct is repeating on multiple test cases which makes it a candidate for a test util (in the `testutils/utils.go` file). Something like: 
```func waitForAlertManagerSendGridReceiver(emailAddresses string[]) ```